bug(web/demo): AtlasChat usage has unknown prop 'chatEndpoint' / 'conversationsEndpoint'

## Summary

`packages/web/src/app/demo/page.tsx:196` passes `chatEndpoint` and `conversationsEndpoint` props to `<AtlasChat>`, but `AtlasChatProps` doesn't define them. `tsgo` surfaces the error consistently:

\`\`\`
src/app/demo/page.tsx(196,11): error TS2322: Type '{ apiUrl: string; apiKey: string; chatEndpoint: string; conversationsEndpoint: string; sidebar: true; }' is not assignable to type 'IntrinsicAttributes & AtlasChatProps'.
  Property 'chatEndpoint' does not exist on type 'IntrinsicAttributes & AtlasChatProps'.
\`\`\`

This is pre-existing (surfaced while running \`tsgo\` in the web package during #1595). At runtime the extra props are silently ignored, so the demo is either pointing at the wrong endpoints today or the prop names drifted and no one noticed.

## Context

- File: \`packages/web/src/app/demo/page.tsx:190-199\`
- Relevant prop shape: \`AtlasChatProps\` in \`packages/web/src/ui/components/atlas-chat\`
- The root \`bun run type\` excludes \`packages/web\`, so this only surfaces when running \`tsgo\` inside the web package directly — that's why it slipped past CI.

## Fix direction

Decide whether demo needs custom endpoints at all (historically demo routed through \`/api/v1/demo/*\`); if yes, add them to \`AtlasChatProps\` and thread them through; if no, drop the props and rely on \`apiUrl\`.

While in the area: add \`packages/web\` to the root \`type\` check (or run its \`tsc\` in CI) so future drift gets caught.

## Labels

\`bug\`, \`area: web\`

--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
     "build:docs": "bun run --filter '@atlas/docs' build",
     "start": "sh scripts/start.sh",
     "lint": "eslint packages/ ee/",
-    "type": "bun run --filter '@useatlas/types' build && bun run --filter '@useatlas/plugin-sdk' build && bun run --filter '@useatlas/sdk' build && tsgo --noEmit",
+    "type": "bun run --filter '@useatlas/types' build && bun run --filter '@useatlas/plugin-sdk' build && bun run --filter '@useatlas/sdk' build && bun run --filter '@useatlas/react' build && tsgo --noEmit && bun run --filter '@atlas/web' type",
     "atlas": "bun packages/cli/bin/atlas.ts",
     "mcp": "bun packages/mcp/bin/serve.ts",
     "db:up": "bash scripts/db-up.sh",

--- a/packages/web/package.json
+++ b/packages/web/package.json
@@ -7,7 +7,7 @@
     "dev": "next dev --turbopack",
     "build": "next build",
     "start": "next start",
-    "type": "tsc --noEmit",
+    "type": "tsgo --noEmit",
     "test": "bun scripts/test-isolated.ts"
   },
   "exports": {

